File: algorithms/287.Find the Duplicate Number/Find_the_Duplicate_Number.py
class Solution(Thread):
def findDuplicate(self, nums):
    """
    :type nums: List[int]
    :rtype: int
    """

    slow = nums[0]
    fast = nums[nums[0]]
    while (slow != fast):
        slow = nums[slow]
        fast = nums[nums[fast]]

    fast = 0;
    while (fast != slow):
        fast = nums[fast]
        slow = nums[slow]

    return slow

File: algorithms/287.Find the Duplicate Number/Find_the_Duplicate_Number.cc
<<<<<<< HEAD
class Solution {
int findDuplicate3(vector<int>& nums)
{
    if (nums.size() > 1)
    {
        int slow = nums[0];
        int fast = nums[nums[0]];
        while (slow != fast)
        {
            slow = nums[slow];
            fast = nums[nums[fast]];
        }

        fast = 0;
        while (fast != slow)
        {
            fast = nums[fast];
            slow = nums[slow];
        }
        return slow;
    }
    return -1;
}
=======
class Solution {
int findDuplicate3(vector<int>& nums)
{
    if (nums.size() > 1)
    {
        int slow = nums[0];
        int fast = nums[nums[0]];
        while (slow != fast)
        {
            slow = nums[slow];
            fast = nums[nums[fast]];
        }

        fast = 0;
        while (fast != slow)
        {
            fast = nums[fast];
            slow = nums[slow];
        }
        return slow;
    }
    return -1;
}
>>>>>>> 6200c8704614e918c8bfa5357c648dd1b4f7eb74
};